src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs

```diff
@@ -191,17 +191,45 @@ public static class StandardLearnersCatalog
         }
 
         /// <summary>
-        /// Predict a target using a linear binary classification model trained with the AveragedPerceptron trainer.
+        /// Predict a target using a linear binary classification model trained with averaged perceptron trainer.
         /// </summary>
+        /// <remarks>
+        /// Perceptron is a classification algorithm that makes its predictions based on a linear function.
+        /// For instance with feature values f0, f1,..., f_D-1, the prediction is given by the sign of sigma[0, D-1] (w_i * f_i), where w_0, w_1,..., w_D-1 are the weights computed by the algorithm.
+        ///
+        /// Perceptron is an online algorithm, i.e., it processes the instances in the training set one at a time.
+        /// The weights are initialized to be 0, or some random values. Then, for each example in the training set, the value of sigma[0, D-1] (w_i * f_i) is computed.
+        /// If this value has the same sign as the label of the current example, the weights remain the same.If they have opposite signs,
+        /// the weights vector is updated by either subtracting or adding (if the label is negative or positive, respectively) the feature vector of the current example,
+        /// multiplied by a factor 0 &lt; a &lt;= 1, called the learning rate.In a generalization of this algorithm, the weights are updated by adding the feature vector multiplied by the learning rate,
+        /// and by the gradient of some loss function (in the specific case described above, the loss is hinge-loss, whose gradient is 1 when it is non-zero).
+        ///
+        /// In Averaged Perceptron (AKA voted-perceptron), the weight vectors are stored,
```

Comment:
Can you give links to anything that talks about this specific method?

Reply:
+1

---
In reply to: [255264150](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255264150) [](ancestors = 255264150)

Reply:
there's already two links at the bottom about perceptron. I couldn't find a definitive link for averaged perceptron. only course handouts or other ml frameworks documentation, referencing the same wiki article. do you have a link in mind?

---
In reply to: [255266671](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255266671) [](ancestors = 255266671,255264150)